contributing-docs/14_node_environment_setup.rst

```diff
@@ -18,6 +18,64 @@
 Node.js Environment Setup
 =========================
 
+Contributing to the UI in Airflow
+-----------------------------------------
+
+In Airflow 3, we are moving the UI away from Flask App Builder views to a pure React powered frontend living at ``airflow/ui``.
+During 3.0 development, we will need to run both the new and legacy UIs at the same time until the new UI is feature-complete.
+But we want to limit modifications to the legacy ``airflow/www`` views, to mainly three rules:
+
+1. Bug fixes to cherry pick for 2.10.x and 2.11
+2. The minimum necessary to unblock other Airflow 3.0 feature work
+3. Fixes to react views which we haven't migrated over yet, but can still be ported over to the new UI
+
+Custom endpoints for the UI will also be moved away from ``airflow/www/views.py`` and to ``airflow/api_ui``.
+Contributions to the legacy views file will follow the same rules.
+Committers will exercise their judgement on what endpoints should exist in the public ``airflow/api_connexion`` versus the private ``airflow/api_ui``
+
+Airflow UI
+-----------------------------------------
+
+``airflow/ui`` is our React frontend powered by Vite with dependencies managed by pnpm.
```

Comment:
What's Vite? Is it https://vitejs.dev/ ?

Reply:
Also asked myself about this and first thought I'm too stupid. THanks Kaxil for asking :-D (I was too shy)

Reply:
Added a link and description.